Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentMode.cpp

```diff
@@ -121,6 +132,741 @@ namespace PhysX
+                            AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                            componentModeCollectionInterface->EnumerateActiveComponents(
+                                [](const AZ::EntityComponentIdPair& entityComponentIdPair, const AZ::Uuid&)
+                                {
+                                    JointsComponentModeRequestBus::Event(
+                                        entityComponentIdPair, &JointsComponentModeRequests::SetCurrentSubMode,
+                                        JointsComponentModeCommon::SubComponentModes::ModeType::Translation);
+                                }
+                            );
+                        }
+                    );
+                }
+            );
+
+            hotKeyManagerInterface->SetActionHotKey(actionIdentifier, "1");
+        }
+
+        // Switch to Rotation Sub-Mode
+        {
+            constexpr AZStd::string_view actionIdentifier = "o3de.action.jointsComponentMode.switchToRotationSubMode";
+            AzToolsFramework::ActionProperties actionProperties;
+            actionProperties.m_name = SubModeData::RotationTitle;
+            actionProperties.m_description = SubModeData::RotationToolTip;
+            actionProperties.m_category = "Joints Component Mode";
+
+            actionManagerInterface->RegisterAction(
+                EditorMainWindowActionContextIdentifier,
+                actionIdentifier,
+                actionProperties,
+                []
+                {
+                    auto componentModeCollectionInterface = AZ::Interface<AzToolsFramework::ComponentModeCollectionInterface>::Get();
+                    AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                    componentModeCollectionInterface->EnumerateActiveComponents(
+                        [](const AZ::EntityComponentIdPair&, const AZ::Uuid&)
+                        {
+                            auto componentModeCollectionInterface =
+                                AZ::Interface<AzToolsFramework::ComponentModeCollectionInterface>::Get();
+                            AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                            componentModeCollectionInterface->EnumerateActiveComponents(
+                                [](const AZ::EntityComponentIdPair& entityComponentIdPair, const AZ::Uuid&)
+                                {
+                                    JointsComponentModeRequestBus::Event(
+                                        entityComponentIdPair, &JointsComponentModeRequests::SetCurrentSubMode,
+                                        JointsComponentModeCommon::SubComponentModes::ModeType::Rotation);
+                                }
+                            );
+                        }
+                    );
+                }
+            );
+
```

Comment:
All this code seems repeated but just changing `JointsComponentModeCommon::SubComponentModes::ModeType::Rotation` part. Maybe that could be moved to a function passing that as parameters and making it reach the lambda with scoped parameters?

Same in other places.

Reply:
Makes sense, will do!

Reply:
+1